contrib/ci/build_freebsd_package.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/sh
+
+GITHUB_SHA=
+GITHUB_REPOSITORY=
+GITHUB_REPOSITORY_OWNER=
+GITHUB_TAG=
+
+while [ -n "$1" ]; do
+    case $1 in
+        --GITHUB_SHA=*)
+            GITHUB_SHA=${1#--GITHUB_SHA=}
+            ;;
+        --GITHUB_REPOSITORY=*)
+            GITHUB_REPOSITORY=${1#--GITHUB_REPOSITORY=}
+            ;;
+        --GITHUB_REPOSITORY_OWNER=*)
+            GITHUB_REPOSITORY_OWNER=${1#--GITHUB_REPOSITORY_OWNER=}
+            ;;
+        --GITHUB_TAG=*)
+            GITHUB_TAG=${1#--GITHUB_TAG=}
+            ;;
+        *)
+            echo "Command $1 unknown. exiting..."
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+if [ -z "$GITHUB_SHA" ] || [ -z "$GITHUB_REPOSITORY" ] || \
+    [ -z "$GITHUB_REPOSITORY_OWNER" ] || [ -z "$GITHUB_TAG" ]; then
+    exit 1
+fi
+
+set -e
+mkdir -p /usr/local/etc/pkg/repos/
+# Fix meson flag problem https://www.mail-archive.com/freebsd-ports@freebsd.org/msg86617.html
+cp /etc/pkg/FreeBSD.conf /usr/local/etc/pkg/repos/FreeBSD.conf
+sed -i .old 's|url: "pkg+http://pkg.FreeBSD.org/${ABI}/quarterly"|url: "pkg+http://pkg.FreeBSD.org/${ABI}/latest"|' \
+/usr/local/etc/pkg/repos/FreeBSD.conf
+pkg install -y meson
+pkg upgrade -y meson
+cd /usr
+git clone https://github.com/3mdeb/freebsd-ports.git --depth 1 -b fwupd ports
```

Comment:
this bit does worry me slightly; can't we get this from freebsd upstream? Ideally this would all be self contained.

Reply:
It will be changed to the upstream repo when fwupd and its dependencies will come into freebsd-ports.